mongodump/mongodump.go

```diff
@@ -946,6 +957,61 @@ func (dump *MongoDump) DumpMetadata() error {
 	return nil
 }
 
+// DumpPreludeMetadata dumps information about the server and the dump in json format
+// Currently only writes the server version and tool version, but we can use this to write other metadata about the dump in the future.
+func (dump *MongoDump) DumpPreludeMetadata() error {
+	preludeJson := map[string]string{
+		"ServerVersion": dump.serverVersion,
+		"ToolVersion":   dump.ToolOptions.VersionStr,
+	}
+
+	filename := "prelude.json"
+
+	if dump.ToolOptions.Namespace.DB != "" {
+		filename = filepath.Join(dump.ToolOptions.Namespace.DB, filename)
+	}
+	if dump.OutputOptions.Out == "" {
+		filename = filepath.Join("dump", filename)
+	} else {
+		filename = filepath.Join(dump.OutputOptions.Out, filename)
+	}
+	if dump.OutputOptions.Gzip {
+		filename += ".gz"
+	}
+	if _, err := os.Stat(filepath.Dir(filename)); err != nil {
```

Comment:
This check probably isn’t necessary. If this directory doesn’t exist, os.Create() below should fail with fs.ErrNotExist, so you can just check for that case.